internal/services/containers/kubernetes_cluster_node_pool_resource.go

```diff
@@ -798,6 +868,13 @@ func resourceKubernetesClusterNodePoolUpdate(d *pluginsdk.ResourceData, meta int
 		props.NetworkProfile = expandAgentPoolNetworkProfile(d.Get("node_network_profile").([]interface{}))
 	}
 
+	if d.HasChange("zones") {
+		zones := zones.ExpandUntyped(d.Get("zones").(*schema.Set).List())
+		if len(zones) > 0 {
```

Comment:
Updated!
As a note in this property (and similar cases mentioned above), i replicated the implementation made on the default_node_pool scenario; not sure if it would have the same problem or the behaviour is different (for whatever reason), maybe that should be verified.